### ISSUE TITLE
codegen: postgres - money data type incorrect map

### ISSUE THREAD

**Issue by RodneyAtRealWineLists** [slick/slick#1005]
codegen 2.1.0

creating the following postgres table:

CREATE TABLE test_ME_money (
        id serial,
        name varchar(300),
        total money
)

generates the following code:

case class TestMeMoneyRow(id: Int, name: Option[String] = None, total: Option[Double] = None)

and i think it should be:

case class TestMeMoneyRow(id: Int, name: Option[String] = None, total: Option[BigDecimal] = None)

when i try and do an insert, my test program fails with the following error when the column is mapped to Double instead of BigDecimal:

[error](run-main-2) org.postgresql.util.PSQLException: ERROR: column "total" is of type money but expression is of type double precision
[error]   Hint: You will need to rewrite or cast the expression.
[error]   Position: 58
org.postgresql.util.PSQLException: ERROR: column "total" is of type money but expression is of type double precision
  Hint: You will need to rewrite or cast the expression.
  Position: 58

**Comment by cvogt** [comment]
probably requires checking what data the model builder puts into the model that triggers the code generator to write a Double

**Comment by ChavXO** [comment]
Float or Double return double in the model builder. Since the monetary type is looks like a float (with two numbers after the decimal point), isn't the problem slightly more nuanced?

**Comment by cvogt** [comment]
good point. Double is not a good type for money. Changing estimate to medium. Could be easy, but requires research on how to properly map this. In fact this may be better places in slick-pg. @tminglei ?

**Comment by tminglei** [comment]
`slick-pg` didn't support pg money yet. But it can be added on demand.

**Comment by cvogt** [comment]
@tminglei what type would you use?

**Comment by tminglei** [comment]
Maybe a custom type, since there isn't an official scala type for this.
Do you have any suggestions?

**Comment by cvogt** [comment]
I don't know the specifics of the postgres money type, so no. Custom works, but it needs to be able to hold the same data.

**Comment by Ir1sh** [comment]
Is there a work around for this? I have a bunch of money columns.  Also the query that is output when the error is thrown runs ok when I run it manually on the db

**Comment by Deliganli** [comment]
+1
data can be queried with plain sql but cannot query it with using slick's method on the classes that slick-codegen produced. Which looks very correct since a Double or BigDecimal cannot express the currency symbol at basic

`Got the exception org.postgresql.util.PSQLException: Bad value for type double : 109,80 TL
java.lang.Exception: Got the exception org.postgresql.util.PSQLException: Bad value for type double : 109,80 TL`

**Comment by allquantor** [comment]
+1
